### ISSUE TITLE
Can no longer precompile assets without database connection after installing Solidus

### ISSUE THREAD

**Issue by nrser** [solidusio/solidus#2202]
With `solidus (2.3.0)` and `solidus_auth_devise (1.6.4)` installed asset compilation crashes without a database connection.

Program flow (as well as I can tell):

    solidus_auth_devise

wants to to draw routes, which calls

    Spree::Auth::Config.draw_frontend_routes

which in turn calls

    Spree::Preferences::StoreInstance#should_persist?

which *crashes* the app if it can't connect to database.

**Is there any reason Solidus needs a database connection during asset compilation?**

If so... why? What database does it need to be connected to? It seems like only the live production database would make any sense for production builds, meaning that Solidus necessitates asset compilation on the live production infrastructure?

This would be a pretty specific need that I haven't seen from any other package we've used, and would seem to defeat much of the purpose of asset compilation, where we compile assets as part of the build process (which is *not* done on the production infrastructure) then package them up into whatever containers needs them, ship them off to CDNs, etc.

If Solidus doesn't *need* to connect to the relevant database to compile assets, it really seems like `Spree::Preferences::StoreInstance#should_persist?` should return `false` if there is no database connection.

### Steps to reproduce

Take your db down and run

    rails assets:precompile

### Expected behavior

Assets should precompile.

### Actual behavior

Process crashes.

### System configuration
**Solidus Version**: 2.3.0

**Extensions in use**: solidus_auth_devise (1.6.4)

**Rails**: 5.1.1
**Ruby**: 2.3.3

**Comment by jhawthorn** [comment]
> Is there any reason Solidus needs a database connection during asset compilation?

This is the result of some technical debt, specifically Spree traditionally stored configuration in the database, including configuration needed at load time. We've slowly been phasing this out and in the upcoming Solidus 2.4.0 the default will finally be changed to not use the database including for extensions (#2112).

In the mean time, you could try adding an initializer to use the (preferred) non-DB preferences:

``` ruby
# config/initializers/solidus_auth_devise.rb
Spree::Auth.config do |config|
  config.use_static_preferences!
end
```

**Comment by aitbw** [comment]
Fixed via #2112
